### PR TITLE
Develop BaseShardedTestCase

### PR DESCRIPTION
**Pull request by alex-dukhno**
@brmeyer Can you review and push this PR?

### CODE REVIEW

File: src/test/java/org/hibernate/shards/test/BaseShardFunctionalTestCase.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	@AfterClassOnce
+	public void endTesting() {
+		if ( session != null
+				&& session.isOpen() ) {
+			session.close();
+		}
+		if ( !ssf.isClosed() ) {
+			ssf.close();
+		}
```
Comment (brmeyer) [inline src/test/java/org/hibernate/shards/test/BaseShardFunctionalTestCase.java:80]: For consistency and completeness, it's usually best to null out session and ssf here as well.

**Comment by brmeyer** [comment]
@Alexander-Dukhno, other than a few really nit-picky comments, things are looking fantastic.  Thanks for your continued help with this!

**Comment by brmeyer** [comment]
@Alexander-Dukhno, doesn't this duplicate BaseShardingIntegrationTestCase to a certain extent?  Should the 2 be combined?

**Comment by alex-dukhno** [comment]
@brmeyer Now, I haven't whole project's picture, for undestand how to adopt tests for **BaseShardedTestCase**. But in future they both will be combined, i am pretty sure :)

**Pull request closed without merging**